src/packages/solid/components/Toggle/Toggle.tsx

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2023 Comcast Cable Communications Management, LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { createMemo, type Component } from 'solid-js';
+import { View, type NodeProps } from '@lightningjs/solid';
+import styles, { type ToggleStyles } from './Toggle.styles.js';
+import type { Tone } from 'types';
+
+export interface ToggleProps extends NodeProps {
+  /**
+   * Indicates whether the Toggle is checked or unchecked.
+   * Setting this to `true` will check the toggle, and setting it to `false` will uncheck it.
+   */
+  checked?: boolean;
+  tone?: Tone;
+  style?: Partial<ToggleStyles>;
+}
+
+const Toggle: Component<ToggleProps> = (props: ToggleProps) => {
+  const positionToggle = (checked: boolean, props: ToggleProps) => {
```

Comment:
positionToggle should handle `undefined` as well, or checked should default to `false`